declarative/README.md

```diff
@@ -159,6 +160,76 @@ Parameters defined by qualifiers (may be an `Optional`, supports `Mappers`):
+3. Second level configuration node is the fully qualified class name of the annotated type (i.e. `com.example.MyType`)
+4. Third level is either name of the feature, or a reserved word `methods`; feature names must be unique within Helidon, third
+   party features may conflict, though that is outside of scope of our rules; methods are only identified by name; if multiple
+   methods with the same name exist, configuration would be used for all of them, or a full signature can be used (i.e.
+   `myMethod(java.lang.String,java.util.List)`), first we look for full signature
+
+See `io.helidon.declarative.codegen.TypeConfigSupport.generateMethodConfig` and
+`io.helidon.declarative.codegen.TypeConfigSupport.generateTypeConfig`.
+
+Now we have a split, that has the same sub-rules - either we configure something under the type (i.e. on same level as `methods`),
+or we configure something under a specific method (i.e. `type-config.fq-class-name.methods.method-name`).
+
+The next configuration level depends on complexity of the feature. WebServer, WebClient will have a dedicated feature config
+node `server` and `client`, with additional configuration option under these; other features may have directly a specific option
+defined here (such as fault tolerance, scheduling) - make sure there is no name conflict between features, if so, a dedicated
+feature node must be created.
+
+Helidon features (existing):
+
+| config key        | Feature         | Notes                                   |
+|-------------------|-----------------|-----------------------------------------|
+| `client`          | WebClient       | may contain `uri` and `web-client` etc. |
+| `server`          | WebServer       | may contain `path`, `listener` etc.     |                     
+| `schedule`        | Scheduling      | CRON or Fixed Rate task                 |
+| `async`           | Fault Tolerance | Asynchronous handler                    |
+| `timeout`         | Fault Tolerance | Timeout handler                         |
+| `retry`           | Fault Tolerance | Retry handler                           |
+| `circuit-breaker` | Fault Tolerance | Circuit Breaker handler                 |
+| `bulkhead`        | Fault Tolerance | Bulkhead handler                        |
+
+Helidon features (planned):
+
+| config key   | Feature     | Notes                                   |
+|--------------|-------------|-----------------------------------------|
+| `rpc-client` | Grpc client | may contain `uri` and `web-client` etc. |
+| `rpc-server` | Grpc server | may contain `listener` etc.             |
+| `security`   | Security    | may contain `roles-allowed` etc.        |
+| `counter`    | Metrics     | Counter metric                          |
+
+Other features must be added as they are being developed
+
+Documented example:
+
+```yaml
+type-config:
+  "com.example.MyType":
+    server:
+      listener: "admin" # override socket name
```

Comment:
(So in Java properties format this is `type-config.com.example.MyType.server.listener = admin`. Is that OK?

Reply:
Yes that is the key that would be used